waku/v2/protocol/waku_archive/driver/postgres_driver.nim

```diff
@@ -0,0 +1,248 @@
+  except DbError:
+    return err("failed to query rows")
+
+  for r in rows:
+    let rowRes = extractRow(r)
+    if rowRes.isErr():
+      return err("failed to extract row")
+
+    results.add(rowRes.get())
+
+  ok(results)
+
+method getMessages*(
+  s: PostgresDriver,
+  contentTopic: seq[ContentTopic] = @[],
+  pubsubTopic = none(PubsubTopic),
+  cursor = none(ArchiveCursor),
+  startTime = none(Timestamp),
+  endTime = none(Timestamp),
+  maxPageSize = DefaultPageSize,
+  ascendingOrder = true
+): ArchiveDriverResult[seq[ArchiveRow]] =
+  var query = """SELECT storedAt, contentTopic, payload, pubsubTopic, version, timestamp, id FROM messages"""
+  var statements : seq[string]
+  var args : seq[string]
+
+  if contentTopic.len > 0:
+    let cstmt = "contentTopic IN (" & "?".repeat(contentTopic.len).join(",") & ")"
+    statements.add(cstmt)
+    for t in contentTopic:
+      args.add(t)
+
+  if pubsubTopic.isSome():
+    statements.add("pubsubTopic = ?")
+    args.add(pubsubTopic.get())
+
+  if cursor.isSome():
+    let comp = if ascendingOrder: ">" else: "<"
+    statements.add("(storedAt, id) " & comp & " (?,?)")
+    args.add($cursor.get().storeTime)
+    args.add($cursor.get().digest.data)
+
+  if startTime.isSome():
+    statements.add("storedAt >= ?")
+    args.add($startTime.get())
+
+  if endTime.isSome():
+    statements.add("storedAt <= ?")
+    args.add($endTime.get())
+
+  if statements.len > 0:
+    query &= " WHERE " & statements.join(" AND ")
+
+  var direction : string
+  if ascendingOrder:
+    direction = "ASC"
+  else:
+    direction = "DESC"
+
+  query &= " ORDER BY storedAt " & direction & ", id " & direction
+
+  query &= " LIMIT ?"
+  args.add($maxPageSize)
+
+  var rows : seq[Row]
+  var results : seq[ArchiveRow]
+  try:
+    rows = s.connection.getAllRows(sql(query), args)
+  except DbError:
+    return err("failed to query rows")
+
+  for r in rows:
+    let rowRes = extractRow(r)
+    if rowRes.isErr():
+      return err("failed to extract row")
+
+    results.add(rowRes.get())
+
+  ok(results)
+
+method getMessagesCount*(s: PostgresDriver): ArchiveDriverResult[int64] =
+  var count : int64
+  try:
+    let row = s.connection.getRow(sql("""SELECT COUNT(1) FROM messages"""))
+    count = parseInt(row[0])
+
+  except DbError:
+    return err("failed to query count")
+  except ValueError:
+    return err("failed to parse query count result")
+
+  ok(count)
+
+
+method getOldestMessageTimestamp*(s: PostgresDriver): ArchiveDriverResult[Timestamp] =
+  return err("not implemented")
+
+method getNewestMessageTimestamp*(s: PostgresDriver): ArchiveDriverResult[Timestamp] =
+  return err("not implemented")
+
+
+method deleteMessagesOlderThanTimestamp*(s: PostgresDriver, ts: Timestamp): ArchiveDriverResult[void] =
+  return err("not implemented")
+
+method deleteOldestMessagesNotWithinLimit*(s: PostgresDriver, limit: int): ArchiveDriverResult[void] =
```

Comment:
I'll have a look whether it needs implementing, otherwise I can do in a separate PR